eogrow/core/area/utm.py

```diff
@@ -67,3 +69,52 @@ def _create_new_split(self, area_geometry: Geometry) -> List[GeoDataFrame]:
             tile_info["total_num"] = len(bbox_list)
 
         return self._to_dataframe_grid(bbox_list, info_list, ["index_n", "index_x", "index_y", "total_num"])
+
+    def transform_grid(self, target_area_manager: AreaManager) -> List[GridTransformation]:
+        """Calculates how grid should be transformed into another area manager."""
+        self._check_target_compatibility(target_area_manager)
+        target_area_manager = cast(UtmZoneAreaManager, target_area_manager)
+
+        source_grid = self.get_grid(add_bbox_column=True)
+        target_grid = target_area_manager.get_grid(add_bbox_column=True)
+
+        source_size = self.config.patch_size_x, self.config.patch_size_y
+        target_size = target_area_manager.config.patch_size_x, target_area_manager.config.patch_size_y
+
+        target_shape, source_shape = zip(*map(reduce_to_coprime, source_size, target_size))
+        source_shape = cast(Tuple[int, int], source_shape)
+        target_shape = cast(Tuple[int, int], target_shape)
+
+        source_grid = self._add_patch_positions(source_grid, self.config, source_shape)
+        target_grid = self._add_patch_positions(target_grid, target_area_manager.config, target_shape)
+
+        return create_transformations(source_grid, target_grid, match_columns=["position_x", "position_y"])
+
+    @staticmethod
+    def _add_patch_positions(grid: List[GeoDataFrame], config: Schema, shape: Tuple[int, int]) -> List[GeoDataFrame]:
+        """Adds columns position_x and position_y that define an absolute position of each patch in a coarser grid."""
+        translate_x = config.patch_buffer_x - config.offset_x
+        translate_y = config.patch_buffer_y - config.offset_y
+        scale_x = config.patch_size_x * shape[0]
+        scale_y = config.patch_size_y * shape[1]
+
+        for gdf in grid:
+            gdf["position_x"] = gdf.BBOX.apply(lambda bbox: round(bbox.min_x + translate_x) // scale_x)
+            gdf["position_y"] = gdf.BBOX.apply(lambda bbox: round(bbox.min_y + translate_y) // scale_y)
+
+        return grid
+
+    def _check_target_compatibility(self, target_area_manager: AreaManager) -> None:
+        """Checks if source and target managers are compatible to define a transformation between them."""
+        if not isinstance(target_area_manager, UtmZoneAreaManager):
+            raise NotImplementedError(
+                f"Grid transformation is only supported into another {self.__class__.__name__}, but got "
+                f"{target_area_manager.__class__.__name__}"
+            )
+
+        for param in ["offset_x", "offset_y"]:
+            if getattr(self.config, param) != getattr(target_area_manager.config, param):
+                raise NotImplementedError(
+                    "Cannot transform grid because source and target area managers have different values of config "
+                    f"parameter {param}."
+                )
```

Comment:
This part is meant to be quite unique for any area manager. They might look similar now but the more transformations we support the more different this will get.